libs/@guardian/source-development-kitchen/src/react-components/ticker/useIsInView.ts

```diff
@@ -0,0 +1,79 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+type Options = {
+	/**
+	 * Defaults to `undefined` (falsy),
+	 * which only trigger on the first intersection.
+	 *
+	 * If `true`, trigger the hook on all intersections.
+	 */
+	repeat?: true;
+	/** Set the initial HTML Element, if known. */
+	node?: HTMLElement;
+};
+
+/**
+ * Custom hook around the `IntersectionObserver`.
+ *
+ * @returns a tuple containing `[isInView, setNode]`
+ */
+const useIsInView = (
```

Comment:
Did we ever create anywhere for lib React hooks to live @sndrs ?

I can't see any other hooks in csnx but this hook will now have slightly different versions in DCR and Source

Reply:
nothing currently, if it's useful `libs` could be a good place?

Reply:
We did wonder about bringing across DCR's version of `useIsInView`. There are a couple of complications:
1. we removed the `debounce` feature, to avoid depending on lodash (and because we don't need it here),
2. it's React-specific, and `@guardian/libs` doesn't currently reference React at all

How do you feel about keeping this alternative version of the hook alongside the ticker for now?

Reply:
I think thats fine for now but would it be possible to add a comment to point to the DCR version please? Something along the lines of:

https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/components/OnwardsUpper.importable.tsx#L23-L28

In terms of how we consolidate, perhaps something we could discuss as a 'client side' group @sndrs ?

Btw we should be able to import `lodash.debounce` as a single dependency which is a relatively small dep.

https://bundlephobia.com/package/lodash.debounce@4.0.8

Reply:
Great! I've added a comment - I'll pop one on the DCR version too.